lib/sockfd_cache.c

```diff
@@ -517,3 +637,45 @@ void sockfd_cache_del(const struct node_id *nid, struct sockfd *sfd)
 	sockfd_cache_del_node(nid);
 	free(sfd);
 }
+
+static void *monitor_sd_node_connectivity(void *ignored)
+{
+	int err;
+
+	sd_info("node connectivity monitor main loop");
+
+	for (;;) {
+		struct sockfd_cache_entry *entry;
+		rb_for_each_entry(entry, &sockfd_cache.root, rb) {
```

Comment:
Not sure why we need to put a read lock here and where should we unlock it. If we unlock after "rb_for_each_entry" block, the sd_write_lock function in prepare_conns() will hang indefinitely.